test/subtype.jl

```diff
@@ -1813,7 +1821,7 @@ let
     # issue #38279
     t = typeintersect(Tuple{<:Array{T, N}, Val{T}} where {T<:Real, N},
                       Tuple{<:Array{T, N}, Val{<:AbstractString}}  where {T<:Real, N})
-    @test t == Tuple{<:Array{Union{}, N}, Val{Union{}}} where N
+    @test t == Tuple{<:Array{T, N}, Val{Union{}}} where {T<:Real, N}
```

Comment:
Could you explain why the answer was wrong before? That means that `Tuple{<:Array{Union{}, N}, Val{Union{}}} where N`is too tight, right? So there is a type that is a subtype of both intersected types but not of the old result? I don't see it. Can you give an example?

Also, given that `(Tuple{<:Array{T, N}, Val{T}} where {T<:Real, N}) == (Tuple{Array{T, N}, Val{T}} where {T<:Real, N})` and `(Tuple{<:Array{T, N}, Val{<:AbstractString}}  where {T<:Real, N}) == (Tuple{Array{T, N}, Val{<:AbstractString}}  where {T<:Real, N})`, this change equally affects `typeintersect(Tuple{Array{T, N}, Val{T}} where {T<:Real, N}, Tuple{Array{T, N}, Val{<:AbstractString}}  where {T<:Real, N})`, right?

Reply:
Yeah I guess this wasn't wrong before after all. And in fact if you remove the redundant typevars, this still gives
```
Tuple{Array{Union{}, N}, Val{Union{}}} where N
```